Frameworks/CoreMotion/CMMotionManager.mm

```diff
@@ -249,56 +321,140 @@ - (void)stopGyroUpdates {
-            _gyroUpdateInterval = self.gyrometer.minimumReportInterval / c_intervalScaleFactor;
+        // iOS uses seconds while WinRT uses milliseconds, hence the multiplication/division by c_secondToMilliseconds
+        if (updateInterval * c_secondToMilliseconds < self.gyrometer.minimumReportInterval) {
+            _gyroUpdateInterval = self.gyrometer.minimumReportInterval / c_secondToMilliseconds;
             NSTraceInfo(TAG, @"gyroUpdateInterval capped to minimum supported value: %d", _gyroUpdateInterval);
         } else {
             _gyroUpdateInterval = updateInterval;
         }
         
-        self.gyrometer.reportInterval = static_cast<unsigned int>(self.gyroUpdateInterval * c_intervalScaleFactor);
+        self.gyrometer.reportInterval = static_cast<unsigned int>(self.gyroUpdateInterval * c_secondToMilliseconds);
     }
 }
 
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)startMagnetometerUpdatesToQueue:(NSOperationQueue*)queue withHandler:(CMMagnetometerHandler)handler {
-    UNIMPLEMENTED();
+    
+    if (self.magnetometerAvailable) {
+        [self startMagnetometerUpdates];
+        self.magnetometerQueue = queue;
+
+        self.magnetometerToken = [self.magnetometer addReadingChangedEvent:
+            ^void(WDSMagnetometer* sender, WDSMagnetometerReadingChangedEventArgs* e) {
+
+                WDSMagnetometerReading* reading = e.reading;
+
+                CMMagnetometerData* data = [[CMMagnetometerData alloc] initWithValues:reading.magneticFieldX 
+                                                                                    y:reading.magneticFieldY
+                                                                                    z:reading.magneticFieldZ
+                                                                                 time:[self toSecondsSinceBoot:reading.timestamp]];
+                [queue addOperationWithBlock:^{ handler(data, nil); }];                                   
+            }];          
+    }    
 }
 
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)startMagnetometerUpdates {
-    UNIMPLEMENTED();
+    
+    @synchronized(self) {
+      
+        if (self.magnetometerAvailable) {
+        
+            if (self.magnetometerActive) {
+                [self stopMagnetometerUpdates];
+            }
+        
+            self.magnetometerActive = true;
+
+            self.magnetometer.reportInterval = static_cast<unsigned int>(self.magnetometerUpdateInterval * c_secondToMilliseconds);
+        } 
+    }               
 }
 
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)stopMagnetometerUpdates {
-    UNIMPLEMENTED();
+    
+    @synchronized(self) {
+
+        if (self.magnetometerActive) {
+            self.magnetometerActive = false;
+        
+            // the reportInterval is set to 0, the default value.
+            self.magnetometer.reportInterval = 0;
```

Comment:
do we need to do this? I'm a little confused on how this will work since 0 obviously < min interval. I understand its a default but I know in other apis, there was some weirdness about not letting you set it back to that.